x-pack/legacy/plugins/siem/public/components/ml/create_influencers.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexItem } from '@elastic/eui';
+import React from 'react';
+import { Anomaly } from './types';
+
+export const createKeyAndValue = (influencer: Record<string, string>): string =>
+  `${Object.keys(influencer)[0]}: “${Object.values(influencer)[0]}”`;
```

Comment:
Doesn't appear to be an issue since this is used within a `map`, but if used elsewhere, influencer could be an empty object without a TS error, and returns `"undefined"`.

Here's a test:

``` ts
  test('it returns expected createKeyAndValue record when empty object is passed', () => {
    const entities = createKeyAndValue({});
    expect(entities).toEqual('');
  });
```

Reply:
I added the above test and added a filter mechanism to the function `createInfluencers` below this so we filter out any empty objects.

Added tests for checking against filter:

```ts
  test('it returns expected createKeyAndValue record when empty object is passed', () => {
    const entities = createKeyAndValue({});
    expect(entities).toEqual('');
  });

  test('it creates the anomalies without filtering anything out since they are all well formed', () => {
    const wrapper = mount(<span>{createInfluencers(anomalies.anomalies[0])}</span>);
    expect(wrapper.text()).toEqual('host.name: “zeek-iowa”process.name: “du”user.name: “root”');
  });

  test('it returns empty text when passed in empty objects of influencers', () => {
    anomalies.anomalies[0].influencers = [{}, {}, {}];
    const wrapper = mount(<span>{createInfluencers(anomalies.anomalies[0])}</span>);
    expect(wrapper.text()).toEqual('');
  });

  test('it filters out empty anomalies but keeps the others', () => {
    anomalies.anomalies[0].influencers = [
      { 'influencer-name-one': 'influencer-value-one' },
      {},
      { 'influencer-name-two': 'influencer-value-two' },
    ];
    const wrapper = mount(<span>{createInfluencers(anomalies.anomalies[0])}</span>);
    expect(wrapper.text()).toEqual(
      'influencer-name-one: “influencer-value-one”influencer-name-two: “influencer-value-two”'
    );
  });
```

Will update any other areas that follow this similar pattern to ensure that if for some odd reason we do get a malformed influencer due to a bug we will safely filter them out.

Appreciate the time you took to write this unit test above for me.